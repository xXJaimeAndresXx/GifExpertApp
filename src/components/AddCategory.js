import React, { useState } from 'react'
import PropTypes from 'prop-types';

export const AddCategory = ({setCategories}) => {

    const [InputValue, setInputValue] = useState('');

    const handleInputChange= (e)=>{
        console.log(e.target.value);
        setInputValue(e.target.value);
    }

    const handleSubmit = (e) => {
        e.preventDefault();
        if (InputValue.trim().length>2){
            setCategories(c=> [InputValue,...c]);
            setInputValue('');

        }
        
        console.log('Submit hecho');

    }

    return (
        
            <form onSubmit= {handleSubmit}>
                
                {/* <h2>Add Category</h2> */}
                <input type= "text" value= {InputValue} onChange={handleInputChange}>

                </input>
            </form>
        
    )
}

AddCategory.propTypes={
    setCategories:PropTypes.func.isRequired
}
