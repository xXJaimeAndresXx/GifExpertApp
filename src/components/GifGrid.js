import React from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs';
// import { getGifs } from '../helpers/getGifs';
import { GifGridItem } from './GifGridItem';
import PropTypes from 'prop-types'


export const GifGrid = ({category}) => {
   
    const {data:images, loading}= useFetchGifs(category);

    // const [images, setImages] = useState([])
    // useEffect(()=>{
    //     getGifs(category)
    //     .then(setImages);
    // },[category])

    
    // getGifs();
    
    return (
        <>
        
        <h3 className= "animate__animated animate__fadeIn">{category}</h3>

        {loading && <p>Loading...</p>}
        
        { <div className="card-grid">
            
          
                    {
                    images.map(img=>(
                        // <li key= {img.id}>{img.title}</li>
                        <GifGridItem
                        key = {img.id}
                        {...img}
                        />
                        
                    ))
                    }
                
            
        </div> }
        </>           
    )
    
}


GifGrid.propTypes={
    category: PropTypes.string.isRequired
}