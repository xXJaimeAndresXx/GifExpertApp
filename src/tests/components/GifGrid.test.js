import {shallow} from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import React from 'react';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');
import '@testing-library/jest-dom'

describe('Pruebas en el GifGrid', () => {

    const category= 'One Punch';
    test('Debe mostrarse correctamente', () => {
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        })

        const wrapper= shallow( <GifGrid category={category}/>);
        expect (wrapper).toMatchSnapshot();
    });

    test('Debe de mostrar items cuando se cargan imagenes con useFetchGifs', () => {
        const gifs=[{
            id: 'ABC',
            url: 'https://hola.ke.ase',
            title: 'Hola'
        }]
        
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        })
        
        const wrapper= shallow( <GifGrid category={category}/>);

        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);

    })
    
    
})
