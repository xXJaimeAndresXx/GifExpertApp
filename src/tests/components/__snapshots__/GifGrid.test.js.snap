// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el GifGrid Debe mostrarse correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    One Punch
  </h3>
  <p>
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
